### ISSUE TITLE
IE Version 0.11-63 - Cloche - Chunk Weirdness and Item build up.

### ISSUE THREAD

**Issue by Mattabase** [BluSunrize/ImmersiveEngineering#2261]
IE Version 0.11-63

https://goo.gl/Z4N4kM

So, this a really odd bug. 

I will test it with just IE shortly. 

So this setup works just fine while someone is in the base, but when I log out and leave the base, come back in about 15 Minutes, the items are piled up like this. 
The base is not being chunk loaded by any means, so nothing should be happening.

**Comment by malte0811** [comment]
Known, I even have a setup in my dev world that sometimes causes this, but I don't have any definite results.
What exactly is build 32? The current builds are something like 62 or 63. 32 was an unreleased 1.10.2 build.

**Comment by Mattabase** [comment]
Sorry

**Comment by Mattabase** [comment]
I am all sorts of bad this morning, Yeah the version number was just a typo...I made twice...

GG Me.

**Comment by malte0811** [comment]
Do you have any sort of chunkloading in the general area? There are a lot of ghostloading issues in IE (and probably most other mods as well), I fixed just those caused by getTileEntity. getBlockState probably causes the same issues.

**Comment by malte0811** [comment]
I did some more testing on this just now: Adding a proper chunkloader to the cloche setup should fix it. You are most likely ghostloading it in some way (maybe some multiblock partially in the chunk or some redstone controlled machine at the chunk border. There are lots of ways to do this).
I looked into the code related to this for a bit, seems like the following happens:
1. Some TE ghostloads the chunk the cloche is in. This includes loading the entities.
2. Two things happen, order unknown. Might even take more than one tick:
 - The TE's in the chunk (cloches) tick, potentially creating entities
 - The entities in that chunk (and probably the rest of it) are unloaded
3. What would happen if the item entities hadn't been unloaded in 2.: The entities would tick, call `onEntityCollided` on the conveyor belts and get moved along nicely. Actually, nothing happens since the entities have been unloaded.

I can see one universal fix on our side (with potential performance issues): have conveyors tick and move the entities using `World#getEntitiesInAABB`. Aside from that pretty much any machine in IE can be used for ghostloading (even after I spend a few hours fixing all `getTileEntity` calls), potentially even some vanilla blocks (`World#isBlockIndirectlyGettingPowered` can cause ghostloading). I have no idea what it is like for other mods.

**Comment by Mattabase** [comment]
Alright, sorry I was slow getting back to this. 

Yeah, I need to look into the bases on the server and find out what the bases may be using that could ghostload things. I know that in the base I have been testing with no active chunkloading is being used.

**Comment by oitsjustjose** [comment]
We have LOTS of Ghost Loading on my server, but again we can't figure it out either. Cloches have crashed our server *twice* today (after being fixed once, fixing again now) due to being in an un-chunkloaded area. I thought it was PersistentBits' fault, but it wasn't seeing as I removed it and the problem persists.

**Comment by BluSunrize** [comment]
...crashlogs?

**Comment by oitsjustjose** [comment]
I would have posted it originally but I didn't figure it'd be particularly helpful; it's the resulting crash when the server creates too many Item Entities to save to the chunk data:

[Log](https://pastebin.com/HN1Q2hee)

**Comment by BluSunrize** [comment]
Is there any other chunkloaders in that area?

**Comment by oitsjustjose** [comment]
No there were not. Our [temporary] solution ended up being adding one.

**Comment by BluSunrize** [comment]
What else was in there that could have caused the ghostloading? Any suspects?

**Comment by oitsjustjose** [comment]
Thermal Dynamics Ducts, a Refined Storage system, and Storage Drawers.

I wonder if it's Storage Drawers - recall that a few months ago I made a report identical to this one and all I was using was IE and Storage Drawers..

**Comment by BluSunrize** [comment]
*shrug*

Might be worth a shot :V

**Comment by HolyDLX** [comment]
I think I have the same issue on our server.

IE Version: Immersive Engineering 0.11-63

### Setup
Got some cloches, a fermenter, squeezer, refinery, dieselgenerator setup in one chunk. Automated with vanilla only.
Don't be confused by the cobble and dirt texture. Just retexturing with chisel. No Chisel&Bits, just Chisel.

![onechunkiepowersetup](https://user-images.githubusercontent.com/4067905/30510324-2cb89f26-9ac2-11e7-8c5f-3bf3367d71d5.png)

The brown floor marks the chunk borders.
Cloches are blocked by redstone signal that is generated from a comparator checking a hopper that pipes from the storage crates into the machines. (Beneath the convayor belts)
On the right we directly go into the storage crate as we only have sugar cane.
On the left we first go into an item router that redirectes the fiber to the top conveyor and into the lava.

Thing is, as you mentioned, I have redstone and redstone sensitive blocks on the chunk border. Also it is next to the chunk that has the end portal (the stone square). Don't know if the endportal does chunkload or something...

If you need any more details let me know.

### Error Description
- When I log off everything is backlogged and idle.
- When I log back in the other day the system stopped running.
- The conveyor belts leading into the squeezer and fermenter are filled with items.
- The fermenter has no items inside and no power.
- The squeezer still has power and a little bit of items inside.
- The refinery ran out of ethanol.
- The backup batteries on the dieselgenerator that power the system are empty.

---

I will probably move the setup one block inwards from the chunk border and see if this fixes the issue. Will report back when I had time to do so and some time passed to get a good result.

**Comment by HolyDLX** [comment]
Moved hoppers and machines away from chunk border. One line of redstone still touched chunk border.

Issue still persists: After logging out for an hour stacks of items are clocked up on the conveyor belts near the cloches. Fermenter was running low on items and emptied out on ethanol.

Will remove line of redstone and test again.

**Comment by HolyDLX** [comment]
Test: Removed line of redstone. Threw item on the ground. Logged back in after an hour.

Expected result: Item did not despawn. Energy was still buffered. No clocked up items in conveyor belts at cloches. No dry fermenter/squeezer.

Actual result: Ethanol completely drained, cloched up items in conveyor belts. Basically a complete system shut down. BUT the cobblestone I threw on the ground is still there.

I really hoped that moving everything away from the chunk borders would fix stuff but apparently not. Will do some last test before putting this project on hold. Will apply a constant redstone signal instead of the comparator and see if maybe the tile entities simply do not respect the redstone signal when in this "ghostloaded" state you wrote about.

**Comment by HolyDLX** [comment]
Applying a permanent redstone signal to the redstone lines via lever stopped the system for good. No power loss or item mess. Tried this several time so far.

---

Will not continue testing for now.

ALso it looks like you just fixed a lot of ghostloading/ticking or whatever in the lastest release (73). To me it looks like this is a MC 1.12 release? Is this going to be ported back to 1.11 or is 1.11 discontinued?

**Comment by BluSunrize** [comment]
1.11 is discontinued, I'm focusing on 1.12 solely currently.
